labrad/test/test_signal.py

```diff
@@ -0,0 +1,66 @@
+import Queue
+
+import pytest
+
+import labrad
+from labrad.server import LabradServer, setting, Signal
+import labrad.util
+
+
+class SignalTestServer(LabradServer):
+    name = "Signal Test Server"
+
+    class_signal = Signal(100, 'class_signal', '?')
+
+    def __init__(self):
+        super(SignalTestServer, self).__init__()
+        self.instance_signal = Signal(200, 'instance_signal', '?')
+
+    @setting(101)
+    def fire_class_signal(self, c, data):
+        self.class_signal(data)
+
+    @setting(201)
+    def fire_instance_signal(self, c, data):
+        self.instance_signal(data)
+
+
+def _test_signals(signal_setting, fire_setting):
+    with labrad.util.syncRunServer(SignalTestServer()):
+        with labrad.connect() as cxn:
+            server = cxn.signal_test_server
+            msg_id = 12345
+
+            # add a listener to enqueue all messages
+            queue = Queue.Queue()
+            def handler(message_ctx, msg):
+                queue.put((message_ctx, msg))
+            cxn._backend.cxn.addListener(handler, ID=msg_id)
+
+            # we don't get messages before signing up
+            server[fire_setting]('not listening')
```

Comment:
In this case, `fire_setting` is a variable containing the string name of the setting we want to call. The settings are accessible by square-bracket lookup like in a dict for situations when you have to compute the name of a setting, or here, where we want to pass the setting name into the function so we can reuse the same code to call different settings.

Note that you can also do lookups on the connection object for servers, if you have the server name in a variable:
```python
server_name = 'my_server'
setting_name = 'my_setting'
cxn[server_name][setting_name](...)

# this is the same as:
cxn.my_server.my_setting(...)
```